### DISCUSSION TITLE
Best practices for custom root CA for all workflows

### DISCUSSION THREAD

**Discussion by sebbbastien** [argoproj/argo-workflows#12483] [Q&A] [answered]
Hi,

I would like to use ArgoWorkflows to build Go binaries, pack them in docker images, and push images to a cloud-based private registry.

Problem is: all HTTPS flows are forced to be inspected by a corporate SWG solution, who implements TLS man-in-the-middle, and sign each TLS communication with an internaly signed and trusted certificate.
Go get step fail complaining about "failed to verify certificate: x509: certificate signed by unknown authority", and I know problem will be the same on docker push step.

One solution can be to declare the internal root CA on _Go build_ and _Docker push_ steps. But I will have to do that on each step for every pipeline/step.

Is there any solution to declare our custom root CA at an upper level, larger level, so each step generated by any pipeline will know this custom root CA?

BR

**Comment by agilgur5** [comment, marked as answer]
Have you tried building a custom image that has the CA in it? That may be the most straightforward way.

You could also use [`templateDefaults`](https://argo-workflows.readthedocs.io/en/latest/template-defaults/) and [`workflowDefaults`](https://argo-workflows.readthedocs.io/en/latest/default-workflow-specs/).
